receiver/awscontainerinsightreceiver/internal/k8swindows/kubelet.go

```diff
@@ -13,82 +13,182 @@ import (
 
 	ci "github.com/open-telemetry/opentelemetry-collector-contrib/internal/aws/containerinsight"
 	cExtractor "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/cadvisor/extractors"
-	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/host"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/k8swindows/extractors"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/stores/kubeletutil"
+
 	"go.uber.org/zap"
 	stats "k8s.io/kubelet/pkg/apis/stats/v1alpha1"
 )
 
-type kubeletSummaryProvider struct {
+// KubeletProvider Represents interface to kubelet.
+type KubeletProvider interface {
+	GetClient() (*kubeletutil.KubeletClient, error)
+	GetSummary() (*stats.Summary, error)
+}
+
+type kubeletProvider struct {
 	logger   *zap.Logger
 	hostIP   string
 	hostPort string
 	client   *kubeletutil.KubeletClient
-	hostInfo host.Info
 }
 
-func new(logger *zap.Logger, info host.Info) (*kubeletSummaryProvider, error) {
-	hostIP := os.Getenv("HOST_IP")
-	kclient, err := kubeletutil.NewKubeletClient(hostIP, ci.KubeSecurePort, logger)
+// GetClient Returns singleton kubelet client.
+func (k *kubeletProvider) GetClient() (*kubeletutil.KubeletClient, error) {
+	if k.client != nil {
+		return k.client, nil
+	}
+	kclient, err := kubeletutil.NewKubeletClient(k.hostIP, k.hostPort, k.logger)
+	if err != nil {
+		k.logger.Error("failed to initialize kubeletProvider client, ", zap.Error(err))
+		return nil, err
+	}
+	k.client = kclient
+	return kclient, nil
+}
+
+// GetSummary Get Summary from kubelet API.
+func (k *kubeletProvider) GetSummary() (*stats.Summary, error) {
+	kclient, err := k.GetClient()
 	if err != nil {
-		return nil, fmt.Errorf("failed to initialize kubelet client: %w", err)
+		k.logger.Error("failed to get kubeletProvider client, ", zap.Error(err))
+		return nil, err
+	}
+
+	summary, err := kclient.Summary(k.logger)
+	if err != nil {
+		k.logger.Error("kubeletProvider summary API failed, ", zap.Error(err))
+		return nil, err
+	}
+	return summary, nil
+}
+
+func createDefaultKubeletProvider(logger *zap.Logger) KubeletProvider {
+	kSP := &kubeletProvider{logger: logger, hostIP: os.Getenv("HOST_IP"), hostPort: ci.KubeSecurePort}
+	return kSP
+}
+
+// kubelet represents receiver to get metrics from kubelet.
+type kubelet struct {
```

Comment:
nit - I don't know if I love the name `kubelet`.  This isn't a kubelet, it is a provider of kubelet metrics

Reply:
Addressed, I changed it to `SummaryProvider`, name of package gives context about summary from kubelet.